### PR TITLE
Skills 

### PR DESCRIPTION
**Pull request by g4s8**
zerocracy/farm#492 - person's skills

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#265](https://github.com/zerocracy/datum/issues/265) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#265](https://github.com/zerocracy/datum/issues/265) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by 0crat** [comment]
@fabriciofx[/z](https://www.0crat.com/u/fabriciofx) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@fabriciofx[/z](https://www.0crat.com/u/fabriciofx) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#265](https://github.com/zerocracy/datum/issues/265), please stop working

**Comment by 0crat** [comment]
This pull request [#265](https://github.com/zerocracy/datum/issues/265) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/431713180). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by carlosmiranda** [comment]
@yegor256 this looks good to merge

**Comment by 0crat** [comment]
To cancel the order use `refuse` as in §6

**Comment by carlosmiranda** [comment]
@yegor256 unassign bug cropped up again...

**Comment by yegor256** [comment]
@carlosmiranda it's not a bug anymore, the ticket hasn't been unassigned, see:

**Comment by yegor256** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/datum/issues/265#issuecomment-365835064))

@yegor256 This is what I know about this job, as in [§32](http://datum.zerocracy.com/pages/policy.html#32):

  * The job [#265](https://github.com/zerocracy/datum/issues/265) is [in scope](http://datum.zerocracy.com/pages/policy.html#14) for 13days
  * The role is `REV`
  * The job is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) for [3days](http://datum.zerocracy.com/pages/policy.html#8)
  * There is a monetary reward attached
  * @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) will get 25% of their hourly rate for this job
  * Some equity will be [vested](http://datum.zerocracy.com/pages/policy.html#37) on completion
  * The job doesn't have any [impediments](http://datum.zerocracy.com/pages/policy.html#9)
  * The [budget](http://datum.zerocracy.com/pages/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @g4s8[/z](https://www.0crat.com/u/g4s8): This user reported the ticket
    * @fabriciofx[/z](https://www.0crat.com/u/fabriciofx): User was resigned from the ticket

File: xsd/pmo/people.xsd [hunk 1]
```diff
@@ -73,6 +73,18 @@ SOFTWARE.
           <xs:restriction base="xs:boolean"/>
         </xs:simpleType>
       </xs:element>
+      <xs:element name="skills">
+        <xs:complexType>
+          <xs:sequence>
+            <xs:element name="skill" minOccurs="0" maxOccurs="unbounded" type="xs:string"/>
```
Comment (yegor256) [inline xsd/pmo/people.xsd]: @g4s8 let's set `maxOccurs` to `4` in order to prevent this list from being too long

**Comment by yegor256** [comment]
@g4s8 one small comment above. Also, let's make sure this `maxOccurs` really work -- create a new test XML for it, please.

**Comment by g4s8** [comment]
@yegor256 done, please see new commit

File: xsd/pmo/people.xsd [hunk 2]
```diff
@@ -73,6 +73,18 @@ SOFTWARE.
           <xs:restriction base="xs:boolean"/>
         </xs:simpleType>
       </xs:element>
+      <xs:element name="skills">
+        <xs:complexType>
+          <xs:sequence>
+            <xs:element name="skill" minOccurs="0" maxOccurs="4" type="xs:string"/>
```
Comment (yegor256) [inline xsd/pmo/people.xsd]: @g4s8 I would also suggest to make it stricter than just `xs:string`. Let's add regex validation: `[a-z0-9+#.-]{2,16}`

**Comment by yegor256** [comment]
@g4s8 one more comment above. In general, the stricter the schema, the more stable is the softwsare.

**Comment by g4s8** [comment]
@yegor256 done, see new commits, but I've made `[a-z]{1,16}` instead of `[a-z]{2,16}` because there are some one-letter languages: `c`, `j` etc.

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), total is [+1335](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+9984](https://www.0crat.com/u/yegor256)

**Pull request merged**